MeshLib/MeshGenerators/VtkMeshConverter.cpp

```diff
@@ -356,8 +358,75 @@ MeshLib::Mesh* VtkMeshConverter::convertUnstructuredGrid(vtkUnstructuredGrid* gr
 		elements[i] = elem;
 	}
 
-	return new MeshLib::Mesh(mesh_name, nodes, elements);
+	MeshLib::Mesh* mesh (new MeshLib::Mesh(mesh_name, nodes, elements));
+	convertScalarArrays(*grid, *mesh);
+
+	return mesh;
+}
+
+void VtkMeshConverter::convertScalarArrays(vtkUnstructuredGrid &grid, MeshLib::Mesh &mesh)
+{
+	vtkPointData &point_data = *grid.GetPointData();
+	unsigned const n_point_arrays = static_cast<unsigned>(point_data.GetNumberOfArrays());
+	for (unsigned i=0; i<n_point_arrays; ++i)
+	{
+		char const*const array_name (point_data.GetArrayName(i));
+		vtkDataArray* array (point_data.GetArray(array_name));
+		convertArray(array, mesh.getProperties(), MeshLib::MeshItemType::Node);
+	}
+
+	vtkCellData &cell_data = *grid.GetCellData();
+	unsigned const n_cell_arrays = static_cast<unsigned>(cell_data.GetNumberOfArrays());
+	for (unsigned i=0; i<n_cell_arrays; ++i)
+	{
+		char const*const array_name (cell_data.GetArrayName(i));
+		if (std::string(array_name).compare("MaterialIDs") == 0)
+			continue;
+		vtkDataArray* array (cell_data.GetArray(array_name));
+		convertArray(array, mesh.getProperties(), MeshLib::MeshItemType::Cell);
+	}
+}
+
+void VtkMeshConverter::convertArray(vtkDataArray* array, MeshLib::Properties &properties, MeshLib::MeshItemType type)
+{
+	vtkIdType const nTuples (array->GetNumberOfTuples());
+	int const nComponents (array->GetNumberOfComponents());
+
+	vtkDoubleArray* double_array = vtkDoubleArray::SafeDownCast(array);
+	if (double_array)
+	{
+		std::vector<double> vec;
+		vec.reserve(nTuples);
+		if (nComponents == 1)
+			for (vtkIdType i=0; i<nTuples; ++i)
+				vec.push_back(double_array->GetComponent(i,0));
```

Comment:
At least for arrays with just 1 component this can be made faster with one of the memory copy options described in [this StackOverflow answer](http://stackoverflow.com/a/261607/80480).